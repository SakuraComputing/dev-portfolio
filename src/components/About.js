import React from 'react';


export const About = () => (
    <section className="about section">
        <div className="section-inner">
            <h2 className="heading">About Me</h2>
            <div className="content">
                <p>Write a brief intro about yourself.</p>
                <p>It's a good idea to include your personal interests and hobbies as well.</p>
            </div>
        </div>
    </section>
);

export default About;
