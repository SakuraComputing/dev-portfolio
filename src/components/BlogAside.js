import React from 'react';

export const BlogAside = () => (
    <aside className="blog aside section">
        <div className="section-inner">
            <h2 className="heading">Latest Blog Posts</h2>
            <p>You can use Sascha Depold's <a href="https://github.com/sdepold/jquery-rss" target="_blank">jQuery RSS
                plugin</a> to pull in your blog post feeds.</p>
            <div id="rss-feeds" className="content">

            </div>
        </div>
    </aside>
);
export default BlogAside;