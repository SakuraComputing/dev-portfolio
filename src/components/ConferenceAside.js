import React from 'react';

export const ConferenceAside = () => (
    <aside className="list conferences aside section">
        <div className="section-inner">
            <h2 className="heading">Conferences</h2>
            <div className="content">
                <ul className="list-unstyled">
                    <li><i className="fa fa-calendar"></i> <a href="https://developer.apple.com/wwdc/" target="_blank">WWDC
                        2014</a> (San Francisco)
                    </li>
                    <li><i className="fa fa-calendar"></i> <a href="http://hive.aigaseattle.org/">Hive</a> (Seattle)
                    </li>
                </ul>
            </div>
        </div>
    </aside>
);

export default ConferenceAside;