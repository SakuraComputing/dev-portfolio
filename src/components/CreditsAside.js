import React from 'react';

export const CreditsAside = () => (

    <aside className="credits aside section">
        <div className="section-inner">
            <h2 className="heading">Technologies Used</h2>
            <div className="content">
                <ul className="list-unstyled">
                    <li><a href="https://reactjs.org/" target="_blank"><i className="fa fa-external-link"></i>React JS</a></li>
                    <li><a href="https://angular.io/" target="_blank"><i className="fa fa-external-link"></i>Angular</a></li>
                    <li><a href="https://nodejs.org/en/" target="_blank"><i className="fa fa-external-link"></i>Node JS</a></li>
                    <li><a href="https://expressjs.com/" target="_blank"><i className="fa fa-external-link"></i>Express</a></li>
                    <li><a href="https://www.mongodb.com/" target="_blank"><i className="fa fa-external-link"></i>Mongo</a></li>
                    <li><a href="http://jquery.com/" target="_blank"><i className="fa fa-external-link"></i> jQuery</a></li>
                    <li><a href="http://getbootstrap.com/" target="_blank"><i className="fa fa-external-link"></i> Bootstrap</a></li>
                    <li><a href="http://sass-lang.com/" target="_blank"><i className="fa fa-external-link"></i> SASS</a></li>
                    <li><a href="http://fortawesome.github.io/Font-Awesome/" target="_blank"><i className="fa fa-external-link"></i> FontAwesome</a></li>
                </ul>
            </div>
        </div>
    </aside>
);

export default CreditsAside;


