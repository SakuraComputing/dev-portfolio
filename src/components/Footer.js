import React from 'react';

export const Footer = () => (
    <footer className="footer">
        <div className="container text-center">
            <small className="copyright">Designed with <i className="fa fa-heart"></i> by <a
                href="/" target="_blank">Xiaoying Riley</a> for developers
            </small>
        </div>
    </footer>
);

export default Footer;