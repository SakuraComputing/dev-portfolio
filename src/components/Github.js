import React from 'react';

export const Github = () => (
    <section className="github section">
        <div className="section-inner">
            <h2 className="heading">My GitHub</h2>
            <div>You can embed your GitHub contribution graph calendar using IonicaBizau's <a
                href="https://github.com/IonicaBizau/github-calendar" target="_blank">GitHub Calendar</a> widget.

                <div id="github-graph" className="github-graph">

                </div>

                <div>You can also embed your GitHub activities using <a
                    href="http://caseyscarborough.com/projects/github-activity/" target="_blank">GitHub Activity
                    Stream</a> widget.
                </div>
                <div id="ghfeed" className="ghfeed">

                </div>
            </div>
        </div>
    </section>
);

export default Github;