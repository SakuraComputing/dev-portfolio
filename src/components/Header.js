import React from 'react';

export const Header = (props) => (

    <header className="header">
        <div className="container">
            <img className="profile-image img-responsive pull-left" src="assets/images/sakura.png" alt="Ewan Bauld"/>
            <div className="profile-content pull-left">
                <h1 className="name">{props.header.name}</h1>
                <h2 className="desc">{props.header.title}</h2>
                <ul className="social list-inline">
                    {props.header.social.twitter.url ? <li><a href={props.header.social.twitter.url}><i className="fa fa-twitter"></i></a></li> : null}
                    {props.header.social.googleplus.url ?<li><a href={props.header.social.googleplus.url}><i className="fa fa-google-plus"></i></a></li> : null}
                    {props.header.social.linkedin.url ? <li><a href={props.header.social.linkedin.url}><i className="fa fa-linkedin"></i></a></li> : null}
                    {props.header.social.github.url ? <li><a href={props.header.social.github.url}><i className="fa fa-github-alt"></i></a></li> : null}
                    {props.header.social.hackernews.url ? <li className="last-item"><a href={props.header.social.hackernews.url}><i className="fa fa-hacker-news"></i></a></li> : null}
                </ul>
            </div>
        </div>
    </header>
);

export default Header;

