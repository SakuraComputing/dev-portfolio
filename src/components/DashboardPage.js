import React from 'react';
import Header from '../components/Header';
import About from '../components/About';
import LatestProject from '../components/LatestProject';
import OtherProjects from '../components/OtherProjects';
import Experience from '../components/Experience';

const DashboardPage = () =>(
    <div>
        <Header />
        <About/>
        <LatestProject/>
        <OtherProjects/>
        <Experience/>
    </div>
);
export default DashboardPage;