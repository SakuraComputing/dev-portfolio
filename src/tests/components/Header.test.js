import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Header } from '../../components/Header';
import json from '../../../data/profile_header';


describe('Header Component', () => {

    let wrapper;

    beforeEach(() => {
        // const header = jest.fn();
        const header = {
            name: "test",
            title: "cc",
            social: {
                linkedin: {
                    url: ""
                }
            },
        };

        wrapper = shallow(<Header header={header} />);
    });

    it('should render header correctly',() => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it('should show linked in account if it exists', () => {
        const header = {
            social: { linkedin: { url: "http://wibble.com" } },
        };
        wrapper = shallow(<Header header={header} />);
        expect(wrapper.find('.fa-linkedin').exists()).toBe(true);
    });
    it('should not show linked in account if it doesnt exist', () => {
        const header = {
            social: { linkedin: { url: "" } },
        };
        wrapper = shallow(<Header header={header} />);
        expect(wrapper.find('.fa-linkedin').exists()).toBe(false);
    });

});

