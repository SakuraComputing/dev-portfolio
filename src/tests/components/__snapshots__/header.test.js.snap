// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render header correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <img
      alt="Ewan Bauld"
      className="profile-image img-responsive pull-left"
      src="/public/images/loader.gif"
    />
    <div
      className="profile-content pull-left"
    >
      <h1
        className="name"
      >
        test
      </h1>
      <h2
        className="desc"
      >
        cc
      </h2>
      <ul
        className="social list-inline"
      >
        <li>
          <a
            href="#"
          >
            <i
              className="fa fa-twitter"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            <i
              className="fa fa-google-plus"
            />
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            <i
              className="fa fa-github-alt"
            />
          </a>
        </li>
        <li
          className="last-item"
        >
          <a
            href="#"
          >
            <i
              className="fa fa-hacker-news"
            />
          </a>
        </li>
      </ul>
    </div>
    <a
      className="btn btn-cta-primary pull-right"
      href="http://themes.3rdwavemedia.com/"
      target="_blank"
    >
      <i
        className="fa fa-paper-plane"
      />
      Contact Me
    </a>
  </div>
</header>
`;
